README.md

```diff
@@ -1,19 +1,1118 @@
-# rialto
+# Rialto Media Framework Documentation
 
-## Project webpage (with current coverage report and Doxygen)
-https://rdkcentral.github.io/rialto/
+Rialto is a modular media framework for RDK Entertainment  platforms that provides an out-of-process media playback solution with a client-server architecture. The framework separates media processing into dedicated server processes, enabling better resource management, security isolation, and system stability for video devices.
 
-## Coding Guidelines:
-https://wiki.rdkcentral.com/display/ASP/Rialto+Coding+Guidelines
+Rialto consists of three main components:
 
-## Commit message template:
-https://wiki.rdkcentral.com/display/ASP/Commit+message+template
+**Rialto Core** - The main framework providing client-server IPC infrastructure, media pipeline services, and server management capabilities. This component forms the backbone of the architecture, managing multiple session servers, coordinating resource allocation, and providing the fundamental APIs for media playback and DRM operations.
 
-## Building and Running Unit Tests:
-https://wiki.rdkcentral.com/display/ASP/Unit+Tests
+**Rialto GStreamer Plugin** - Integration layer that bridges GStreamer-based applications with the Rialto media server. This plugin provides custom GStreamer sink elements (rialtomseaudiosink, rialtomsevideosink, rialtomsesubtitlesink, rialtowebaudiosink) for GStreamer pipelines in web browsers and native applications.
 
-## Building Rialto using Yocto:
-https://wiki.rdkcentral.com/display/ASP/Building+Rialto+using+Yocto
+**Rialto OCDM** - OpenCDM (Content Decryption Module) adapter enabling DRM functionality through the Rialto framework. This component implements the standard OpenCDM interface expected by EME (Encrypted Media Extensions) implementations, routing all DRM operations through Rialto's MediaKeys service for centralized license management and secure content decryption.
 
-## Running Rialto:
-https://wiki.rdkcentral.com/display/ASP/Running+YouTube+with+Rialto
+Rialto enables multiple applications to share hardware media decoders efficiently while maintaining process isolation. The Server Manager orchestrates multiple RialtoSessionServer instances, each dedicated to a specific application, with each session assigned a unique Unix domain socket for IPC communication. The design provides scalability, fault isolation, and resource utilization across the middleware stack. Media processing isolation prevents application crashes from affecting media playback, and system resources are managed based on application priority and state.
+
+**Rialto Framework**
+
+The following diagram shows the Rialto media framework architecture, illustrating the relationships between client applications, the Rialto framework components, and the hardware abstraction layers.
+
+![Rialto Introduction](./docs/Overview.png)
```

Comment:
There's something wrong with the png files. I cannot render them even after downloading them.